### PR TITLE
Update migration guides

### PR DESCRIPTION
**Pull request by chlowell**
Updates migration guides.

### CODE REVIEW

**Comment by rpetrusha** [comment]
@chlowell, are you going to be handling migration from microsoft/dotnet in place of @conniey? 

Also FYI, I have two open PRs -- one that adds two new compatibility topics and one that (hopefully) corrects some formatting issues -- in the microsoft/dotnet repo.

**Comment by chlowell** [comment]
I'm working on automation to update migration guides when doc changes merge in microsoft/dotnet and am using this PR to look at the rendered output. It's mostly equivalent to what's live now, but I notice some xrefs don't resolve:
 - `System.Web.Http.RouteAttribute`
 - `System.Windows.Controls.VirtualizingStackPanel.SetIsVirtualizing`
    - These are in the MSDN documentation but not the new API browser; are their assemblies not indexed?
 - `System.Windows.Forms.MonthCalendar.MonthCalendarAccessibleObject`
    - Type doesn't exist; should it be `System.Windows.Forms.MonthCalendar.AccessibilityObject`?
 - `System.Windows.Controls.Primitives.Selector.SelectionChanged%2A`
    - The problem here seems to be the encoded asterisk (`%2A`)

**Comment by rpetrusha** [comment]
I'm taking these out of order:
-  System.Windows.Controls.VirtualizingStackPanel.SetIsVirtualizing: this is an attached property, which currently API browser doesn't support. It should sometime in the (near) future, so although it doesn't resolve to a link, it should be left as is; it will at some point.
- System.Windows.Forms.MonthCalendar.MonthCalendarAccessibleObject: this should be AccessibilityObject. I'll add it to my existing PR.
- System.Web.Http.RouteAttribute.ctor(System.String): I don't know why this doesn't work. In my PR, I changed the \# to %23, its escape code. But I'm not sure that will accomplish anything.
- System.Windows.Controls.Primitives.Selector.SelectionChanged%2A: the problem may be the %2A, though this should work. In the PR, I've removed the %2A.

When you do the next migration, we can see if three of these four issues have been resolved.

**Comment by chlowell** [comment]
Looks like that helped, but still some failures due to encoding. Is there documentation somewhere of precisely what needs to be encoded, and by what scheme?

**Comment by rpetrusha** [comment]
What are the failures, @chlowell. Generally, affected APIs seem to require asset IDs, but everything else is passed through to the open publishing system and requires \<xref> syntax.

**Comment by chlowell** [comment]
`System.Windows.Forms.MonthCalendar.AccessibilityObject` and `System.Web.Http.RouteAttribute.%23ctor` don't resolve. Otherwise, I think this looks good.

**Comment by rpetrusha** [comment]
@chlowell, the problem with AccessibilityObject is that the property is inherited from Control. I'll open a PR to fix it, but I'm not fully sure how API references in the Affected APIs section are resolved. If we use asset IDs, we have to do it as \`T:System.Windows.Forms.Control.AccessibilityObject\`, which is confusing. If we can do it as a link and xref, we can do it as \[System.Windows.Forms.MonthCalendar.AccessibilityObject\](xref:System.Windows.Forms.Control.AccessibilityObject). I'll open a PR that changes the link to the latter, and if you migrate it, we can see how it builds in your PR. If it fails, then we'll have to use the asset ID.

**Comment by mairaw** [comment]
And the RouteAttribute only exists on MSDN as part of the Web API 2 set. We can add an xref link to the https://github.com/dotnet/docs/blob/master/_zip/missingapi.yml file.

**Comment by rpetrusha** [comment]
Also, @chlowell, there are a three merge conflicts caused by changes that were made to migration topics directly in the dotnet/docs repo. I'm not sure whether the migration routine changes URLs, though, so I'm not sure that relative URLs will resolve properly when migrated. Do you know, by any chance?

**Comment by rpetrusha** [comment]
The immediate problem with System.Web.Http.RouteAttribute, @chlowell and @mairaw, is that I'm not certain whether the correct type is System.Web.Http.RouteAttribute or System.Web.Mvc.RouteAttribute; I suspect the latter is more likely. (The first is largely undocumented.) I'll try to see if I can find the original bug that formed the basis of this PR. Unfortunately, the link to Microsoft Connect no longer works -- it redirects to a general page that says Connect has been retired. We'll have to remove it.
I'll open a PR in Microsoft/dotnet to fix the WinForms link and remove the link to Microsoft Connect; we can leave it open until the RouteAttribute issue is resolved.

**Comment by chlowell** [comment]
@rpetrusha the logic currently copies URLs in the Microsoft/dotnet source documents directly. I don't have any special knowledge of the right thing to do with them, I'm assuming they're problematic because #4374 changes some to relative URIs (also I see some of the URLs include a locale).

**Comment by rpetrusha** [comment]
As long as it copies them directly, these changes should all be OK, @chlowell. I'll resolve the merge conflicts so that this PR builds. I'm about to open a PR in the Microsoft/dotnet repo for the changes, also. The correct type is System.Web.Mvc.RouteAttribute. I'll also open a PR to modify the missingapi.yml file, @mairaw.

**Comment by chlowell** [comment]
@rpetrusha Looks good to me, `System.Windows.Controls.VirtualizingStackPanel.SetIsVirtualizing` is the only remaining broken xref.

**Comment by mairaw** [comment]
There are several other warnings in the [build report](https://opbuildstorageprod.blob.core.windows.net/report/2018%5C2%5C23%5Ca3bda507-3390-de91-8f7f-26f90f4e5fc8%5CPullRequest%5C201802232146423577-4288%5Cworkflow_report.html?sv=2015-02-21&sr=b&sig=S0E2sOp1AaqgRL7xf%2Bch4TeGo%2FeCWT%2FpyFYk7jSlug8%3D&st=2018-02-23T22%3A23%3A20Z&se=2018-03-26T22%3A28%3A20Z&sp=r) @chlowell. Our current number of warnings on master is 9.

`System.Windows.Controls.VirtualizingStackPanel.SetIsVirtualizing` doesn't seem like the right API. We can add however the API for https://msdn.microsoft.com/en-us/library/system.windows.controls.virtualizingstackpanel.isvirtualizing on the missing mapping file since attached properties are still missing from docs, if that's the one you're looking for.

**Comment by chlowell** [comment]
Thanks for pointing that out, there are new broken links in the source docs. I've opened Microsoft/dotnet#632 to fix those.

I think your link for `VirtualizingStackPanel.IsVirtualizing` is appropriate. Mapping the broken links to it sounds like the best we can do until attached properties are in the new docs.

File: includes/migration-guide/retargeting/networking/default-value-servicepointmanagersecurityprotocol.md [hunk 1]
```diff
@@ -2,7 +2,7 @@
...
...ramework 4.7, the default value of the <xref:System.Net.ServicePointManager.SecurityProtocol?displayProperty=nameWithType> property is <xref:System.Net.SecurityProtocolType.SystemDefault?displayProperty=nameWithType>. This change allows .NET Framework networking APIs based on SslStream (such as FTP, HTTPS, and SMTP) to inherit the default security protocols from the operating system instead of using hard-coded values defined by the .NET Framework. The default varies by operating system and any custom configuration performed by the system administrator. For information on the default SChannel protocol in each version of the Windows operating system, see [Protocols in TLS/SSL (Schannel SSP)](https://msdn.microsoft.com/library/windows/desktop/mt808159.aspx).For applications that target an earlier version of the .NET Framework, the default value of the <xref:System.Net.ServicePointManager.SecurityProtocol?displayProperty=nameWithType> property depends on the version of the .NET Framework targeted. See the [Networking section of Retargeting Changes for Migration from .NET Framework 4.5.2 to 4.6](~/docs/framework/migration-guide/retargeting/4.5.2-4.6.md#networking) for more information.|
```
Comment (conniey) [inline includes/migration-guide/retargeting/networking/default-value-servicepointmanagersecurityprotocol.md]: There should be a space after `[Protocols in TLS/SSL (Schannel SSP)](https://msdn.microsoft.com/library/windows/desktop/mt808159.aspx).For applications` 
=> `[Protocols in TLS/SSL (Schannel SSP)](https://msdn.microsoft.com/library/windows/desktop/mt808159.aspx). For...`

File: includes/migration-guide/retargeting/winforms/applicationfiltermessage-no-longer-throws-for-re-entrant-implementations.md [hunk 2]
```diff
@@ -3,7 +3,7 @@
...
+|Suggestion|Be aware that <xref:System.Windows.Forms.Application.FilterMessage(System.Windows.Forms.Message@)> will no longer throw for the re-entrant <xref:System.Windows.Forms.IMessageFilter.PreFilterMessage(System.Windows.Forms.Message@)> behavior described above. This only affects applications targeting the .NET Framework 4.6.1.Apps targeting the .NET Framework 4.6.1 can opt out of this change (or apps targeting older Frameworks may opt in) by using the [DontSupportReentrantFilterMessage](~/docs/framework/migration-guide/mitigation-custom-imessagefilter-prefiltermessage-implementations.md#mitigation) compatibility switch.|
```
Comment (conniey) [inline includes/migration-guide/retargeting/winforms/applicationfiltermessage-no-longer-throws-for-re-entrant-implementations.md]: ` .NET Framework 4.6.1.Apps targeting the` should have a space. => `.NET Framework 4.6.1. Apps targeting the`

**Review by conniey — APPROVED** [review]

**Comment by rpetrusha** [comment]
@chlowell, should the WIP label be removed? I think that this is close -- with the latest PR that you opened in Microsoft/dotnet and once the IsVirtualizing corrections are made, this should be ready to go.

File: xml/System/Uri.xml [hunk 3]
```diff
@@ -730,11 +726,7 @@ http://myUrl/
   
 ## Remarks  
  This constructor creates a new <xref:System.Uri> instance by combining an absolute <xref:System.Uri> instance, `baseUri`, with a relative <xref:System.Uri> instance, `relativeUri`. If `relativeUri` is an absolute <xref:System.Uri> instance (containing a scheme, host name, and optionally a port number), the <xref:System.Uri> instance is created using only `relativeUri`.  
- 
- If the `baseUri` has relative parts (like `/api`), then the relative part must be terminated with a slash, (like `/api/`), if the relative part of `baseUri` is to be preserved in the constructed <xref:System.Uri>. 
-
- Additionally, if the `relativeUri` begins with a slash, then it will replace any relative part of the `baseUri`
- 
+
```
Comment (pkulikov) [inline xml/System/Uri.xml]: Why does it look as if the changes of PR #4431 would be reverted upon the merge into master?

File: samples/snippets/csharp/how-to/strings/ParseStringsUsingSplit.cs [hunk 4]
```diff
@@ -101,10 +97,10 @@ private static void SplitUsingStrings()
...
             System.Console.WriteLine("{0} substrings in text:", words.Length);  
     
-            foreach (var word in words)
-            {
-                System.Console.WriteLine(word);
-            }
+            foreach (string s in words)  
+            {  
+                System.Console.WriteLine(s);  
+            }
```
Comment (pkulikov) [inline samples/snippets/csharp/how-to/strings/ParseStringsUsingSplit.cs]: Why does it look as if the changes of PR #4467 would be reverted upon the merge into master? Or I misunderstand something?
The same comment is true for the `docs/csharp/how-to/parse-strings-using-split.md` file changes.

**Comment by chlowell** [comment]
@rpetrusha The content is ready but I need to rebase, the WIP label should stay so no one merges before I've done that.

File: samples/snippets/csharp/how-to/strings/ParseStringsUsingSplit.cs (hunk 4 above)
Comment (chlowell) [reply samples/snippets/csharp/how-to/strings/ParseStringsUsingSplit.cs]: I need to rebase before this merges.

**Comment by chlowell** [comment]
After adding a missing API mapping for `VirtualizingStackPanel.IsVirtualizing` there are no publishing issues with the files changed in this PR. I think it's ready to review and merge.

**Comment by rpetrusha** [comment]
@chlowell, in looking at the published output, there are a fairly large number of things that need to be changed in the microsoft/dotnet repo. In particular, a large number of uuids (e.g., T:\<fully qualified type name>) are not resolving, so they appear as uuids in the output.

Since this PR improves on the current version of the app compat docs, my inclination is to approve and merge it. I can then make the corrections this week, in which case you'd have to do a new migration and open a new PR after I merge my PR. Or do you want to keep this PR open while I make the changes, remigrate content from the microsoft/dotnet repo, and then merge it?

**Comment by chlowell** [comment]
I also lean toward merging this PR, provided none of the changes regress the docs (I haven't seen any change that does so). Migrating again in a week is no problem, and the next PR should be less noisy because the migration guides probably won't need changes.

**Review by rpetrusha — APPROVED** [review]
I've just spot-checked this, but it looks good -- it's added no broken links to the publishing report, it incorporates some changes (including a misclassification of some app compat issues) and so improves on the existing app compat doc set. I'll approve and merge this now, @chlowell. Let's plan on a new migration and PR next week.

**Pull request merged**